lib/client/multiplex.js

```diff
@@ -45,10 +45,10 @@ module.exports = function multiplex(Spark) {
     var mp = this;
     this.conn.on('data', function ondata(data) {
       if (isArray(data)) {
-        var type = data.shift()
-          , id = data.shift()
-          , name = data.shift()
-          , payload = data.shift()
+        var type = data[0]
+          , id = data[1]
+          , name = data[2]
+          , payload = data.length === 4 ? data[3] : null
```

Comment:
Sure. The whole step looks stupid I know, but oob access slows down v8 considerably, increasing the benchmark time by about 50%. Pretty surprising tbh since it should be perfectly safe to do.